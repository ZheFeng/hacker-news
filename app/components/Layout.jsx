import React, { PropTypes } from 'react';
import { Link } from 'react-router';


function nav(text, to) {
  return (
    <Link to={to}>
      {
        ({ isActive, href, onClick }) => (
          <li className={`nav-item ${isActive ? 'active' : ''}`}>
            <a className="nav-link" onClick={onClick} href={href}>
              {text}
            </a>
          </li>
        )
      }
    </Link>
  );
}
const Layout = ({ children }) => (
  <div style={{ paddingTop: '56px' }}>
    <nav
      className="navbar navbar-toggleable-md navbar-inverse bg-primary fixed-top"
    >
      <Link className="navbar-brand" to="/">Hacker News</Link>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          {nav('News', '/newstories')}
          {nav('Best', '/beststories')}
          {nav('Ask', '/askstories')}
          {nav('Show', '/showstories')}
          {nav('Job', '/jobstories')}
        </ul>
      </div>
    </nav>
    <br />
    <div className="container-fluid">
      {children}
    </div>
  </div>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};
export default Layout;
