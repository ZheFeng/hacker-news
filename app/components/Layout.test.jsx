/* eslint-disable import/no-extraneous-dependencies, max-nested-callbacks */
/* global describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router';

import Layout from './Layout';

describe('<Layout />', () => {
  it('Snapshot Test', () => {
    const component = renderer.create(
      <BrowserRouter><Layout>Test</Layout></BrowserRouter>,
  );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
