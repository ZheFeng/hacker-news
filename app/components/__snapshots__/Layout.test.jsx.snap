exports[`<Layout /> Snapshot Test 1`] = `
<div>
  <nav
    className="navbar navbar-toggleable-md navbar-inverse bg-primary">
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
      style={Object {}}>
      Hacker News
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNav">
      <ul
        className="navbar-nav">
        <li
          className="nav-item ">
          <a
            className="nav-link"
            href="/newstories"
            onClick={[Function]}>
            News
          </a>
        </li>
        <li
          className="nav-item ">
          <a
            className="nav-link"
            href="/beststories"
            onClick={[Function]}>
            Best
          </a>
        </li>
        <li
          className="nav-item ">
          <a
            className="nav-link"
            href="/askstories"
            onClick={[Function]}>
            Ask
          </a>
        </li>
        <li
          className="nav-item ">
          <a
            className="nav-link"
            href="/showstories"
            onClick={[Function]}>
            Show
          </a>
        </li>
        <li
          className="nav-item ">
          <a
            className="nav-link"
            href="/jobstories"
            onClick={[Function]}>
            Job
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <br />
  <div
    className="container-fluid">
    Test
  </div>
</div>
`;
